hibernate-core/src/main/java/org/hibernate/persister/entity/JoinedSubclassEntityPersister.java

```diff
@@ -1416,7 +1423,7 @@ public void pruneForSubclasses(TableGroup tableGroup, Map<String, EntityNameUse>
 		// In addition, we switch from a possible LEFT join, to an INNER join for all tablesToInnerJoin
 		if ( innerJoinOptimization ) {
 			final TableReferenceJoin[] oldJoins = tableReferenceJoins.toArray( new TableReferenceJoin[0] );
-			tableReferenceJoins.clear();
+			tableGroup.clearTableReferenceJoins();
```

Comment:
IMO this change makes it harder to follow the rest of the code, since the other code also operates (adds/removes) on the `tableReferenceJoins` collection.

Reply:
I added it because it took me a while to understand how the `TableGroup#tableJoins` list at some point was empty